design/select/index.jsx

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState, useRef } from 'react'
+import classNames from 'classnames'
+
+import classNameHelpers from '../class-name-helpers.css'
+import styles from './styles.css'
+
+// TODO: Taken from @oacore/design
+const generateId = () =>
+  Math.random()
+    .toString(36)
+    .substr(2, 9)
+
+const KEYS = {
+  ESC: 27,
+  ENTER: 13,
+  LEFT: 37,
+  UP: 38,
+  RIGHT: 39,
+  DOWN: 40,
+}
+
+// TODO: Consider converting it to class component
+const Select = ({
+  onSelectionChange,
+  value,
+  options,
+  className,
+  tag: Tag = 'div',
+  id = generateId(),
+}) => {
+  const inputRef = useRef(null)
+  const [searchTerm, setSearchTerm] = useState('')
+  const [showSuggestions, toggleShowSuggestions] = useState(false)
+  const [activeSuggestion, setActiveSuggestion] = useState({})
+  const [loading, setLoading] = useState(false)
+  const [suggestions, setSuggestions] = useState([])
+
+  const handleKeyDown = event => {
+    let pos = suggestions.findIndex(s => s.id === activeSuggestion.id) || 0
+
+    switch (event.which) {
+      case KEYS.ESC:
+        inputRef.current.blur()
+        toggleShowSuggestions(false)
+        break
+
+      case KEYS.RIGHT:
+        inputRef.current.blur()
+        onSelectionChange(activeSuggestion)
+        break
+
+      case KEYS.ENTER:
+        event.preventDefault()
+        event.stopPropagation()
+        if (activeSuggestion.id !== undefined)
+          onSelectionChange(activeSuggestion)
+
+        inputRef.current.blur()
+        break
+
+      case KEYS.UP:
+        event.preventDefault()
+        event.stopPropagation()
+        if (suggestions.length === 0) return
+
+        if (activeSuggestion.id === undefined) {
+          setActiveSuggestion(suggestions[0])
+          return
+        }
+
+        pos = pos <= 1 ? 0 : pos - 1
+        setActiveSuggestion(suggestions[pos])
+        break
+
+      case KEYS.DOWN:
+        event.preventDefault()
+        event.stopPropagation()
+        if (suggestions.length === 0) return
+
+        if (activeSuggestion.id === undefined) {
+          setActiveSuggestion(suggestions[0])
+          return
+        }
+
+        pos = pos >= suggestions.length ? suggestions.length : pos + 1
+        setActiveSuggestion(suggestions[pos])
+        break
```

Comment:
It can be shorter:

```js
case KEYS.UP:
case KEYS.DOWN:
  event.preventDefault()
  event.stopPropagation()
  if (suggestions.length === 0) return

  if (activeSuggestion.id === undefined) {
    setActiveSuggestion(suggestions[0])
    return
  }

  const direction = event.which - KEYS.DOWN + 1 // either -1 or 1
  pos = (pos + direction) % suggestions.length
  setActiveSuggestion(suggestions[pos])
  break
```

Reply:
Fixed here https://github.com/oacore/dashboard/pull/17/commits/dc372d560a95150c7db3d96eca3f4219b97c64ec Thanks